### PR TITLE
WIP: Fixing eager loading mostly

### PR DESCRIPTION
**Pull request by naveed-ahmad**
# TODO
- Fix REST API
- Specs - yes this PR will have functional specs(controller and json response) at-least 
- Benchmarking

### CODE REVIEW

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32955

File: app/models/concerns/language_filterable.rb [hunk 1]
```diff
@@ -12,6 +12,8 @@ module LanguageFilterable
     def self.filter_by_language_or_default(language = 'en')
       language = Language.find_by_id_or_iso_code(language)
 
+      #maybe use where for ONE query
+      # where(language: language).or(where(language: Language.default)
```
Comment (naveed-ahmad) [inline app/models/concerns/language_filterable.rb]: note: order by language priority column, then limit to 1 row for where. Go for it!

File: app/controllers/graphql_controller.rb [hunk 2]
```diff
@@ -18,7 +18,7 @@ def ensure_hash(ambiguous_param)
     case ambiguous_param
     when String
       if ambiguous_param.present?
-        ensure_hash(JSON.parse(ambiguous_param))
+        Oj.load(ambiguous_param)
```
Comment (mmahalwy) [inline app/controllers/graphql_controller.rb]: very nice!

File: app/graphql/types/json_type.rb [hunk 3]
```diff
@@ -3,6 +3,6 @@
 
   description "Parsing JSON"
 
-  coerce_input ->(value, _ctx) { JSON.parse(value) }
-  coerce_result ->(value, _ctx) { JSON.parse(value) }
+  coerce_input ->(value, _ctx) { Oj.load(value) }
```
Comment (mmahalwy) [inline app/graphql/types/json_type.rb]: Very nice!

**Comment by mmahalwy** [comment]
This is awesome! Excited for the benchmarks!

**Pull request closed without merging**